python-checks/src/main/java/org/sonar/python/checks/hotspots/HashingDataCheck.java

```diff
@@ -0,0 +1,195 @@
+    "sha512", "shake_128", "shake_256", "scrypt")
+    .stream()
+    .map(hasher -> "hashlib." + hasher)
+    .collect(Collectors.toSet());
+
+  private static final Set<String> questionablePasslibAlgorithm = immutableSet(
+    "apr_md5_crypt", "argon2", "atlassian_pbkdf2_sha1", "bcrypt",
+    "bcrypt_sha256", "bigcrypt", "bsd_nthash", "bsdi_crypt",
+    "cisco_asa", "cisco_pix", "cisco_type7", "crypt16",
+    "cta_pbkdf2_sha1", "des_crypt", "django_argon2", "django_bcrypt",
+    "django_bcrypt_sha256", "django_des_crypt", "django_disabled",
+    "django_pbkdf2_sha1", "django_pbkdf2_sha256", "django_salted_md5",
+    "django_salted_sha1", "dlitz_pbkdf2_sha1", "fshp", "grub_pbkdf2_sha512",
+    "hex_md4", "hex_md5", "hex_sha1", "hex_sha256", "hex_sha512",
+    "htdigest", "ldap_bcrypt", "ldap_bsdi_crypt", "ldap_des_crypt", "ldap_hex_md5",
+    "ldap_hex_sha1", "ldap_md5", "ldap_md5_crypt", "ldap_pbkdf2_sha1",
+    "ldap_pbkdf2_sha256", "ldap_pbkdf2_sha512", "ldap_plaintext", "ldap_salted_md5",
+    "ldap_salted_sha1", "ldap_sha1", "ldap_sha1_crypt", "ldap_sha256_crypt",
+    "ldap_sha512_crypt", "lmhash", "md5_crypt", "msdcc", "msdcc2",
+    "mssql2000", "mssql2005", "mysql323", "mysql41", "nthash", "oracle10",
+    "oracle11", "pbkdf2_sha1", "pbkdf2_sha256", "pbkdf2_sha512", "phpass", "plaintext",
+    "postgres_md5", "roundup_plaintext", "scram", "scrypt", "sha1_crypt", "sha256_crypt",
+    "sha512_crypt", "sun_md5_crypt", "unix_disabled", "unix_fallback")
+    .stream()
+    .map(hasher -> "passlib.hash." + hasher)
+    .collect(Collectors.toSet());
+
+
+  private static final Set<String> questionableDjangoHashers = immutableSet(
+    "PBKDF2PasswordHasher", "PBKDF2SHA1PasswordHasher", "Argon2PasswordHasher",
+    "BCryptSHA256PasswordHasher", "BasePasswordHasher", "BCryptPasswordHasher", "SHA1PasswordHasher", "MD5PasswordHasher",
+    "UnsaltedSHA1PasswordHasher", "UnsaltedMD5PasswordHasher", "CryptPasswordHasher")
+    .stream()
+    .map(hasher -> "django.contrib.auth.hashers." + hasher)
+    .collect(Collectors.toSet());
+
+
+  @Override
+  public Set<AstNodeType> subscribedKinds() {
+    return immutableSet(
+      PythonGrammar.CALL_EXPR,
+      PythonGrammar.ATTRIBUTE_REF,
+      PythonGrammar.ATOM,
+      PythonGrammar.EXPRESSION_STMT,
+      PythonGrammar.CLASSDEF);
+  }
+
+  @Override
+  public void visitNode(AstNode node) {
+    if (node.is(PythonGrammar.ATTRIBUTE_REF, PythonGrammar.ATOM)) {
+      checkQuestionableHashingAlgorithm(node);
+
+    } else if (node.is(PythonGrammar.EXPRESSION_STMT)) {
+      checkOverwriteDjangoHashers(node);
+
+    } else if (node.is(PythonGrammar.CLASSDEF)) {
+      checkCreatingCustomHasher(node);
+
+    } else if (node.is(PythonGrammar.CALL_EXPR)) {
+      super.visitNode(node);
+    }
+  }
+
+  @Override
+  protected boolean isException(AstNode callExpression) {
+    if (getQualifiedName(callExpression).equals("django.contrib.auth.hashers.make_password")) {
```

Comment:
I would extract all this to a dedicated method, like "isDjangoMakePasswordFunction", to make it easy to read.
Maybe having a simple line of javadoc explaining why we need to exclude this function would be nice as well
```
  @verride
  protected boolean isException(AstNode callExpression) {
    return isDjangoMakePasswordFunction(callExpression);
  }
```